JINSU/Domino/Assets/CodeScript/ObjectCreator.cs

```diff
@@ -0,0 +1,103 @@
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+
+public class ObjectCreator : MonoBehaviour
+{
+
+
+    private ObjectManager objectManager;
+
+    //public으로 변경후 단일씬에 하나씩 스크립트를 gameobject형태로 넣는것을 추천하긴함.
+    //책임분리와 유연성이 있다고하며, 동적생성의 이점을 상실하는 이유
+    //이렇게 등록을 하게되면, 여기에서 실제 프리팹(obj를 인스턴스로 생성)
+
+    //대신에 따로 분리하게된다면 씬의 특성에 맞는 ObjectCreator를 만들어야한다는 데요?
+    private int selectedPrefabIndex = 0;
+
+    void Awake()
+    {
+        objectManager = ObjectManager.Call; //초기에 할당해줘버림.
+        if (objectManager == null)
+        {
+            Debug.LogError("ObjectManager instance is not found.");
+        }
+        objectManager.prefabs.Add(LoadPrefabs("Default_Invisible_ball", "InvisibleBall"));
+        objectManager.prefabs.Add(LoadPrefabs("Default_Domino_block", "Domino"));
+        objectManager.prefabs.Add(LoadPrefabs("Default_Map_plane", "Plane"));
+
+        DontDestroyOnLoad(gameObject);
+
+    }
+
+    void OnDestory()
+    {
+
+    }
+    //TODO
+    //Prefab의 Layer에는 등록이안되는? -> 이후 도미노와 벽간의 충돌시 문제발생 가능성
+    private GameObject LoadPrefabs(string prefabsName, string objName)
+    {
+        GameObject obj = (GameObject)Resources.Load("Prefabs/" + prefabsName);
+        obj.name = objName;
+        objectManager.RegisterObject(obj, obj.name);
+        return objectManager.GetObject(obj.name);
+    }
+    private GameObject CreateObject(string prefabsName, string objName, Vector3 position, Quaternion rotation)
```

Comment:
```suggestion
    private GameObject CreateObject(string objName, Vector3 position, Quaternion rotation)
```
`string prefabsName` 가 사용되지 않는 것 같은데 삭제해도 될 것 같습니다

Reply:
수정시에 잘못된 부분이 있던것 같은데 다시 보고 수정하겠습니다.